File: get_the_offer/18-链表中倒数第k个结点/code.js
function FindKthToTail(head, k) {
    let right = head;
    for (let i = 0; i < k; ++i) {
        if (right === null) {
            // 链表长度小于k
            return null;
        }
        right = right.next;
    }

    let left = head;
    while (right) {
        left = left.next;
        right = right.next;
    }

    return left;
}

File: get_the_offer/18-链表中倒数第k个结点/code.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @param {Integer} k
# @return {ListNode}
def get_kth_from_end(head, k)
    slow = fast = head

    (k-1).times do
        fast = fast.next
    end

    while fast.next
        slow = slow.next
        fast = fast.next
    end

    return slow
end
